account-kit/react/src/components/auth/types.ts

```diff
@@ -9,4 +13,18 @@ export type AuthType =
       placeholder?: string;
     }
   | { type: "passkey" }
-  | { type: "external_wallets"; walletConnect?: WalletConnectParameters };
+  | { type: "external_wallets"; walletConnect?: WalletConnectParameters }
+  | ({ type: "social"; scope?: string; claims?: string } & (
+      | {
+          authProviderId: "auth0";
+          isCustomProvider?: false;
+          auth0Connection?: string;
+          logoUrl: string;
+        }
+      | {
+          authProviderId: KnownAuthProvider;
+          isCustomProvider?: false;
+          auth0Connection?: never;
```

Comment:
if we add `logoUrl?: never` here then you won't have to do the `logoUrl in ...` checks above